bindings/c/foundationdb/fdb_c.h

```diff
@@ -313,10 +335,21 @@ DLLEXPORT WARN_UNUSED_RESULT fdb_error_t fdb_future_readbg_get_descriptions(FDBF
                                                                             FDBBGFileDescription** out,
                                                                             int* desc_count);
 
+/* FIXME: deprecate and remove non _713 versions before 71.3 release is cut */
 DLLEXPORT WARN_UNUSED_RESULT FDBResult* fdb_readbg_parse_snapshot_file(const uint8_t* file_data, int file_len);
 
 DLLEXPORT WARN_UNUSED_RESULT FDBResult* fdb_readbg_parse_delta_file(const uint8_t* file_data, int file_len);
 
+DLLEXPORT WARN_UNUSED_RESULT FDBResult* fdb_readbg_parse_snapshot_file_713(const uint8_t* file_data,
```

Comment:
If fdb_readbg_parse_delta_file and fdb_readbg_parse_delta_file are not released yet, why can't we simply change their signature?

Otherwise, we should not use snowflake specific version numbers, if we want to keep the same name, then better add just some more generic suffix like "_v2"

Reply:
The main reason is that we have done some prerelease development that depends on these.
I can ask if we can simply change the signature but it may break downstream builds within snowflake temporarily.
v2 is probably simpler anyway though so i can change to that

Reply:
Can the API version be used for this? That's ordinarily how we'd change an API

Reply:
I guess probably not if they're unreleased

Reply:
Yeah I think API version is how we'd want to evolve them in the future.
My hope was basically that, since there is the prerelease dependency here, to just add these new ones, update the dependent code, and then delete the old ones before cutting the next 71.3 release

Reply:
Can we change Snowflake code with the same change that updates the FDB header? If not we should think of some more robust change management approach for prerelease testing, because we want nice function names for the released API. Snowflake is not the only one using this header.

Reply:
Do you mean putting the deprecated versions of the functions under something like #ifdef FDB_PRERELEASE_730rc1 in this PR and then defining it in the Snowflake code before actually updating the header?

Reply:
How would we specify that header guard if this code is open source and the snowflake code is not?

Reply:
Since this is a prerelease only concern, the idea would be to have something like this:

```
#ifdef USE_UPDATED_BG_FUNCTIONS
  void my_function(int x) {
    my_function_v2(x);
  }
#else
  void my_function() {
    my_function_v1();
  }
#endif
```

Then in the application code that we are intending to change, this would continue to work with the old functions until the application adds the following line to their code:

```
#define USE_UPDATED_BG_FUNCTIONS
#include "fdb_c.h"
```

Then when all of the application code has been updated and before the release, we can update the FDB C code to remove the header guards:

```
void my_function(int x);
```

The applications can remove the `#define USE_UPDATED_BG_FUNCTIONS` at their leisure after that point.

The benefit of this approach is that the end result is a function named what we want it to be named (i.e. no `v2` suffix) without requiring us to change application code multiple times.

Reply:
This sounds like the best plan, thanks AJ! I'll implement that

Reply:
After offline discussion - the dependencies are not too hard to modify and behind build flags, so we can just change the function signatures and they will be fixed promptly downstream.

Reply:
Sounds good